enes-homework-week3-moreJS.js

```diff
@@ -0,0 +1,200 @@
+  if (code === 1){
+    vehicles = "car";
+  } else if (code === 2){
+    vehicles = "motorcycle";
+  } else {
+    vehicles = "";
+    console.log("please select 1 for cars and 2 for motorcycles");
+  }
+}
+function vehicleType(color, code) {
+  codeDesigner(color, code);
+  console.log("a " + color + " " + vehicles);
+}
+
+vehicleType("blue", 1);
+
+  /* Another method: 
+  (However I could not find out how to write down  
+    "please select 1 for cars and 2 for motorcycles" 
+    when selected other than 1,2 with this method.)
+  function vehicleType(color, code) {
+    vehicles = (code ===1 ) ? "car" :
+              (code === 2 ) ? "motorcyle" : 
+              "";
+  console.log("a " + color + " " + vehicles);
+  }
+  */
+
+vehicleType("blue", 1);
+
+
+//Q5
+
+console.log((3===3) ? "yes" : "no"); //Since .log is an function it is possible to write down an expression in it.
+
+//Q6 
+/* function vehicle() necessitates the first two functions.
+let vehicles = "";
+function codeDesigner(color, code) {
+  if (code === 1){
+    vehicles = "car";
+  } else if (code === 2){
+    vehicles = "motorcycle";
+  } else {
+    vehicles = "";
+    console.log("please select 1 for cars and 2 for motorcycles");
+  }
+}
+function vehicleType(color, code) {
+  codeDesigner(color, code);
+  console.log("a " + color + " " + vehicles);
+}
+*/
+let situation = "";
+function vehicle(color, code, age){
+  codeDesigner(color, code);
+  situation = (age == 0) ? "brand-new" : "used";
+  console.log("a " + color + " " + situation + " " + vehicles);
+}
+
+vehicle("gold", 2, 5);
+
+//Q7
+
+let listOfVehicles = ["motorbike", "caravan", "bike", "plane", "yacht"];
+console.log(listOfVehicles); // expected output: motorbike caravan bike plane yacht  
+
+//Q8
+
+console.log(listOfVehicles[2]); // expected output: bike
+
+//Q9 
+let vehicles = "";
+function codeDesigner(color, code){
+  vehicles = type = (code === 1) ? "motorbike" : 
+         (code === 2 ) ? "caravan" :
+         (code === 3 ) ? "bike" :
+         (code === 4 ) ? "plane" :
+         (code === 5 ) ? "yacht" : "";
+}
+
+
+vehicle("green", 3, 1);
+
+//Q10
+
+let joesGarage = ["car", "motorbike", "caravan", "bike", "plane", "yacht"];
+let arr;
+//let arr2 = [""] when I defined arr2 here, in the question 11 it throw out error. 
+function advertisementMaker(arr){
+  let arr2 = [""];
+  for (let i = 0; i < arr.length; i++){
+    arr2[i] = arr[i] + "s"; 
+  }
+  arr2 = arr2.slice(0, arr2.length - 1).join(', ') + ", and " + arr2.slice(-1);
+  console.log("Amazing Joe's Garage, we service " + arr2 + ".");
+}
+
+advertisementMaker(joesGarage);
+
+//Q11
+
+joesGarage.push("tank");
+advertisementMaker(joesGarage);
+//normally it would work. but somehow, the functions works in the question 10 but not in question 11. I have solved this problem buy defining an array which is valid only inside the function.
```

Comment:
Hi Enes, could you please elaborate on this a bit more? What do you mean that the function doesn't work for Q11 if you declared `arr2` inside the function?

Reply:
When I declared arr2 outside of the function advertisementMaker(arr), the function works only for once. When you add new item to the array on the next line and recall function again, the function would not work.  Because arr2 points some data in it because of the first action. Then I put this arr2 inside the function the problem resolved.

Reply:
Yes. That is because earlier `arr2` was a sort-of global variable and hence when we modified it inside the function, it affected all instances of the variable. Which is why I suggested in our class to keep your functions pure(don't update any variable outside the function's ownership) and this is something you will be hearing often. 🙂